test/interpolating_tests.jl

```diff
@@ -166,18 +175,33 @@ function analytical_derivative(p, t)
 end
 
 integrand_values = IntegrandValues(Vector{Float64})
+integrand_values_inplace = IntegrandValues(Vector{Float64})
 function callback_saving_linear(u, t, integrator, sol)
     return [-sol(t)[2] 0; 0 sol(t)[1]]' * u
 end
+function callback_saving_linear_inplace(du, u, t, integrator, sol)
+    du .= [-sol(t)[2] 0; 0 sol(t)[1]]' * u
```

Comment:
```suggestion
    mul!(du, [-sol(t)[2] 0; 0 sol(t)[1]]', u)
```

would save an allocation, but still has one of course.